lib/routes/security_pin/lock_screen.dart

```diff
@@ -30,9 +31,10 @@ class _AppLockScreenState extends State<AppLockScreen> {
   @override
   Widget build(BuildContext context) {
     return WillPopScope(
-      onWillPop: () {
-        return Future.value(widget.canCancel);
-      },
+      onWillPop: willPopCallback(
```

Comment:
Maybe it is simpler just to set onWillPop to `Future.value(true)` in case widget.canCancel is true? Then we don't need to propagate canCancel.

Reply:
I did add the function looking for this issue https://github.com/breez/breezmobile/issues/461 so I can write something like this:

```
canCancel: () {
  if (drawer.isOpen) {
    drawer.close();
    return false;
  } else {
    return true;
  }
```

So both cases will look similar and keep the close logic in the same place.

To get the same but with the check outside of the `willPopCallback` function, I will make something like that:

```
onWillPop: () {
  if (drawer.isOpen) {
    drawer.close();
    return Future.value(false);
  } else {
    return willPopCallback(context).call();
  }
```

To me it is ok despite of being more verbose.

So what do you think @roeierez ? Change the check to be outside the `willPopCallback` function or keep it there?

Reply:
I see, if we have two cases already then encapsulating that the way you did looks good to me.